source/crud/crud.rst

```diff
@@ -672,7 +672,10 @@ As documented above, the only supported option is maxTimeMS::
   ]
 
 Similar to the count command, the estimated count of documents is returned
-in the ``n`` field.
+in the ``n`` field. Implementations can assume that the document containing
+the single result of the aggregation pipeline is contained in the first batch of
+the server's reply to the aggregate command. It is not necessary to execute a getMore
+operation to ensure that the result is available.
```

Comment:
I didn't follow the entire conversation you had with @jyemin in Slack, but why is this a concern? We don't have any language like this for [countDocuments](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#countdocuments), which also uses an aggregation pipeline. In both cases, we're `$group`-ing into a single bucket. Is there any conceivable reason to expect that the single result document would _not_ be in the immediate command response to `aggregate`?

Reply:
> Why is this a concern?

It's a concern because the most natural way to implement estimatedDocumentCount (and the way both Benji and I did it without really thinking about the ramifications) is to check out a connection, check the wire version, and then send either a count or an aggregate command.  But once you go down that road, it's actually quite challenging from a design perspective to fully iterate any cursor returned in the reply to aggregate, and neither of our implementations attempted to do so.  countDocuments is different because there is no wire version check required.

> Is there any conceivable reason to expect that the single result document would not be in the immediate command response to aggregate?

Yes, because there is nothing documented anywhere that requires a server to return any results in the first batch.  The fact that it does so could be considered an implementation detail of MongoDB.  That's why we want someone from server query team to sign off on this assumption.

Reply:
Your response made me wonder if there's a lingering concern for retryable reads, and possibly needing to recreate the command on a second attempt if the wire version changes. Is that outstanding, or would you say it's already covered by [3b. Sending an equivalent command for the second attempt](https://github.com/mongodb/specifications/blob/master/source/retryable-reads/retryable-reads.rst#b-sending-an-equivalent-command-for-the-second-attempt)?